torch/csrc/utils/disable_torch_function.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+#include <torch/csrc/python_headers.h>
+
+namespace torch {
+  bool torch_function_enabled();
+}
+
+typedef struct {
+    PyObject_HEAD
+    /* Type-specific fields go here. */
+    bool old_state;
+} DisableTorchFunction;
+
+static PyObject* DisableTorchFunction__enter(PyObject* self, PyObject *unused);
+static PyObject* DisableTorchFunction__exit(PyObject* self, PyObject *unused);
```

Comment:
These functions aren't linked because they're declared `static`. You should move everything into `disable_torch_function.cpp` and just export a function that calls `PyType_Ready` to create the type.